doc/developer/platform/ux.md

```diff
@@ -905,6 +905,12 @@ Materialize maintains four correctness guarantees.
    Materialize sends an acknowledgement for (*e.g., the commit of an offset by a
    Kafka source, the acknowledgement of an LSN by a PostgresSQL source, etcetera*),
    will be saved in durable storage.
+5. *Materialize respects the compaction frontier of its sources.* This means
+   that on source creation Materialize queries and puts a read hold on the
+   `initial_since` frontier of the upstream system and sets the initial since
+   *of the collection in Materialize* to be `{reclock(t) for t in
+   initial_since}`. Informally, this prevents users from reading a collection
+   for a source where the contents describe a state that never happened.
```

Comment:
Might be worth explicitly discussing that not all sources support read holds, and in that case Materialize just has to make its best attempt.

Reply:
Given that having upstream compaction run ahead under your feet causes correctness issues what should sources do when their best attempt fails? In other words, should the kafka source immediately emit a permanent error if it detects upstream compaction went beyond its read cursor?

Reply:
Is that even detectable? I was just thinking that the correct property should leave itself an exemption for sources that don't support read holds, and then we can figure out what the best effort behavior is on a source by source basis.

"Materialize respects the compaction frontier of its sources, to the extent permitted by the upstream system."

Reply:
> Is that even detectable? 

Depends on the source. A postgres source with a non-zero `max_slot_wal_keep_size` will advance the since frontier past us if we stay down long enough and it is detectable. For Kafka I'm not sure. I'll rephrase it to leave it up to source implementations although it would be nice to have a preferred behavior under the assumption that it is detectable.